src/utils/api.ts

```diff
@@ -0,0 +1,104 @@
+const apiUrl = "https://disease.sh/v3/covid-19/";
+
+async function request(endpoint: string) {
+  const response = await fetch(apiUrl + endpoint);
+  return response.json();
+}
+
+export async function getAllCountryData() {
+  const countries = await request(`countries`);
+  countries.forEach((e) => {
+    e.locationName = e.country;
+  });
+  return countries;
+}
+
+export async function getOverviewData(counrty: string) {
+  let result, outdated;
+  if (counrty === "all") {
+    result = request(`all?yesterday=true`);
+    outdated = true;
+  } else {
+    result = await request(`countries/${counrty}`);
+    if (result.todayCases == 0) {
+      result = request(`countries/${counrty}?yesterday=true`);
+      outdated = true;
+    } else {
+      outdated = false;
+    }
+  }
+  return Object.assign(result, { outdated });
```

Comment:
I forgot the precise distinction between `Object.assign` and spread, but can this be rewritten as

```suggestion
  return { ...result, ...outdated };
```

or something like that?

Reply:
Indeed. Much better.